Core/tests/ExpressionCodeGenerator.cpp

```diff
@@ -392,9 +396,123 @@ TEST_CASE("ExpressionCodeGenerator", "[common][events]") {
               "123) ?? \"\"");
     }
   }
-  SECTION("Function name") {
+  SECTION("Properties (1 level)") {
+    gd::PropertiesContainer propertiesContainer(gd::EventsFunctionsContainer::Extension);
+
+    auto projectScopedContainersWithProperties = gd::ProjectScopedContainers::MakeNewProjectScopedContainersForProjectAndLayout(project, layout1);
+    projectScopedContainersWithProperties.AddPropertiesContainer(propertiesContainer);
+
+    propertiesContainer.InsertNew("MyProperty");
+    propertiesContainer.InsertNew("MyProperty2");
+
+    gd::EventsCodeGenerator codeGeneratorWithProperties(platform, projectScopedContainersWithProperties);
+
+    {
+      auto node =
+        parser.ParseExpression("MyProperty + 1");
+      gd::ExpressionCodeGenerator expressionCodeGenerator("number",
+                                                          "",
+                                                          codeGeneratorWithProperties,
+                                                          context);
+
+      REQUIRE(node);
+      node->Visit(expressionCodeGenerator);
+      REQUIRE(expressionCodeGenerator.GetOutput() == "getPropertyMyProperty() + 1");
+    }
+    {
+      auto node =
+        parser.ParseExpression("MyProperty + MyProperty2");
+      gd::ExpressionCodeGenerator expressionCodeGenerator("number",
+                                                          "",
+                                                          codeGeneratorWithProperties,
+                                                          context);
+
+      REQUIRE(node);
+      node->Visit(expressionCodeGenerator);
+      REQUIRE(expressionCodeGenerator.GetOutput() == "getPropertyMyProperty() + getPropertyMyProperty2()");
+    }
+  }
+  SECTION("Scene variables (1 level)") {
```

Comment:
Are there cases to show what is generated in case of name collisions?

Reply:
I added one for object <> scene variable (object is used in priority)
I added one for scene <> global variable (scene is used in priority), and fixed a bug because the priority was inverted :)